### PR TITLE
Fix contour plot

### PR DESCRIPTION
**Pull request by liangz1**
Signed-off-by: Liang Zhang <liang.zhang@databricks.com>

## What changes are proposed in this pull request?

The contour plot is incorrect, the x, y, z, and tooltips don't make sense.
The data passed into the contour plot is mis-sorted. They don't really need to be sorted. This PR removed the sorting code and the plot shows correct values.

## How is this patch tested?

Manually tests.
Before:
Screen Shot 2022-01-27 at 4 12 51 PM

After:
Screen Shot 2022-01-27 at 4 15 56 PM

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js [hunk 1]
```diff
@@ -182,9 +171,9 @@ export class CompareRunContour extends Component {
           data={[
             // contour plot
             {
-              z,
-              x: xsSorted,
-              y: ysSorted,
+              z: zs,
+              x: xs,
+              y: ys,
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js:176]: Does this really work?

Comment (liangz1) [reply mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js:176]: I attached a plot, it looks correct to me.

Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js:176]: Ah I see they don't need to be sorted.

**Comment by harupy** [comment]
https://plotly.com/javascript/reference/contour/ indicates that `z` must be a 2D array, but it seems outdated.

> The data from which contour lines are computed is set in `z`. Data in `z` must be a 2D array of numbers. Say that `z` has N rows and M columns, then by default, these N rows correspond to N y coordinates (set in `y` or auto-generated) and the M columns correspond to M x coordinates (set in `x` or auto-generated). By setting `transpose` to "true", the above behavior is flipped.

File: mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js [hunk 2]
```diff
@@ -150,23 +150,12 @@ export class CompareRunContour extends Component {
...
-    xs.forEach((_, index) => {
-      const xi = xsSorted.indexOf(xs[index]);
-      const yi = ysSorted.indexOf(ys[index]);
-      z[yi][xi] = zs[index];
-    });
-
     const maybeRenderPlot = () => {
       const invalidAxes = [];
-      if (xsSorted.length < 2) {
+      if ([...new Set(xs)].length < 2) {
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js:155]:
```suggestion
      if (new Set(xs).size < 2) {
```

File: mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js [hunk 3]
```diff
@@ -150,23 +150,12 @@ export class CompareRunContour extends Component {
...
         invalidAxes.push('X');
       }
-      if (ysSorted.length < 2) {
+      if ([...new Set(ys)].length < 2) {
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js:158]:
```suggestion
      if (new Set(ys).size < 2) {
```

**Review by harupy — APPROVED** [review]
Left minor comments, LGTM!

**Comment by liangz1** [comment]
@harupy https://plotly.com/javascript/reference/contour/#contour-connectgaps It's said that when z is 1d array, the missing data is filled in by default.

In the test here:

https://github.com/plotly/plotly.js/blob/9d8b737cba118e181635ca72326054bb49ccbcb5/test/jasmine/tests/contour_test.js#L77-L83

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @liangz1 !

**Pull request merged**